### DISCUSSION TITLE
Third party libraries

### DISCUSSION THREAD

**Discussion by robsontenorio** [livewire/livewire#7065] [1. Help]
### Import
```blade
{{--  AutoNumeric.js  --}}
<script src="https://cdnjs.cloudflare.com/ajax/libs/autonumeric/4.8.1/autoNumeric.min.js"></script>
```

### Logic
```php
public string $amount = '13';

public function save()
{
   $this->amount = rand();
}
```

### This works and it fills the input with a random number.

```blade
<div>
    Input: <input id="my-input" value="{{ $amount }}" />
    <br>
    livewire: {{ $amount }}
    <br>
    <input type="hidden" wire:model="amount" />
</div>

<button wire:click="save">Save</button>
```

<img width="287" alt="image" src="https://github.com/livewire/livewire/assets/118955/5575a4f3-29e2-431d-8caf-43e21188be19">

<br>

### Once I add this and refresh the page (command+R / F5)....

```html
[...]

<button wire:click="save">Save</button>

<script>
    var element = new AutoNumeric('#my-input');
</script>
```
**The input gets formatted**
<img width="313" alt="image" src="https://github.com/livewire/livewire/assets/118955/ab84cb5c-cf66-455e-b22e-2fb416357e84">

### But... when I click save the input field won't be filled with a random number. 

<img width="319" alt="image" src="https://github.com/livewire/livewire/assets/118955/83c868f3-5701-4d21-95e2-1333ba565c01">

<br>

### What I have tried 

- No luck with `wire:ignore` https://livewire.laravel.com/docs/wire-ignore 
- No luck with `x-int`

```blade
<div x-init="new AutoNumeric($refs.myInput)">
    Input: <input x-ref="myInput" value="{{ $amount }}" />
    [...]
</div>
```

### Works with RANDOM `wire:key`... Why? Should I worry about any side effects?

```html
<div wire:key="money-{{ rand() }}" x-init="new AutoNumeric($refs.myInput)">
    Input: <input x-ref="myInput" value="{{ $amount }}" />

   <!-- PREVIOUS CODE HERE -->
</div>
````

**Comment by lrljoe** [comment]
I suggest you share the output from your browser, and explain where you're putting these script references
